apps/channels/tests/test_forms.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+from django.forms.widgets import HiddenInput, Select
+
+from apps.channels.forms import ChannelForm
+from apps.channels.models import ChannelPlatform
+from apps.service_providers.models import MessagingProvider, MessagingProviderType
+from apps.utils.factories.service_provider_factories import MessagingProviderFactory
+
+
+@pytest.mark.parametrize("platform,", ["whatsapp", "telegram"])
+def test_channel_form_reveals_provider_types(team, platform):
+    """Test that the message provider field is being hidden when not applicable to a certain platform"""
+    # First create a messaging provider
+    message_provider = MessagingProviderFactory(type=MessagingProviderType("twilio"), team=team)
+    MessagingProviderFactory(type=MessagingProviderType("twilio"))
+
+    form = ChannelForm(initial={"platform": ChannelPlatform(platform)}, team=team)
+    widget = form.fields["messaging_provider"].widget
+    form_queryset = form.fields["messaging_provider"].queryset
+    if platform == "whatsapp":
+        assert isinstance(widget, Select)
+    else:
+        assert isinstance(widget, HiddenInput)
```

Comment:
instead of having an if statement here you could pass the expected type in to the test function:

```python
@pytest.mark.parametrize("platform,widget", [
    ("whatsapp", Select),
    ("telegram", HiddenInput),
])
def test_channel_form_reveals_provider_types(team, platform, widget):
```

Reply:
https://github.com/dimagi/open-chat-studio/pull/130/commits/c395dd250a2e7941fef29fbd3869faa253cff18e